virtcontainers/kata_agent.go

```diff
@@ -451,6 +451,21 @@ func (k *kataAgent) startSandbox(sandbox *Sandbox) error {
 		hostname = hostname[:maxHostnameLen]
 	}
 
+	//check grpc server is serving
+	if err := k.connect(); err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	if _, err := k.client.Check(ctx, &grpc.CheckRequest{}); err != nil {
+		k.Logger().Error("grpc server check failed")
```

Comment:
How about:

```
k.Logger().WithError(err).Error("grpc server check failed")
```

Reply:
This log output is used to locate the problem from kata-runtime logfile when kata-agent is crashed in VM

Reply:
Sure, but your code is not logging the actual error, so I'm suggesting you keep your error message and also add the original error (which might include more detail).

Reply:
Thanks! This is a great suggestion about the error log.Back to your first comment,  I agree with your opinion that  kata-agent isn't running is very small likelihood . But I have another question, do we have some measures to make sure that kata-agent will not fail in VM?